DataDog/datadog-operator
# [profiles][CECO-1085] Add priority class name to dap

### What does this PR do?

Add `priorityClassName` as a configurable field in DAP. This can override the `priorityClassName` in the node agent

### Motivation

Blocker for internal usage

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

1. Enable DAP in the operator
2. Create DDA and DAP without `priorityClassName` to ensure it still works
3. Add `priorityClassName` to the DAP and check that the agent pod uses it (`kubectl describe pod xxx`)
4. Test that changing the `priorityClassName` in the DAP updates the agent pod
5. Remove the `priorityClassName` in the DAP and check that it's removed from the agent pod

### Checklist
